providers/azure.go

```diff
@@ -54,11 +59,21 @@ var (
 		Path:   "/v1.0/me",
 	}
 
+	// Default Group URL for Azure.
+	// Pre-parsed URL of https://graph.microsoft.com/v1.0/me.
+	azureDefaultGroupURL = &url.URL{
```

Comment:
Although you're using v2 endpoints, by scoping your token for the graph endpoint you are limiting yourself to 1.0 tokens.
Have you verified generating v2.0 tokens works correctly? The claims are different and do not contain all the same values as v1 (for instance `email` is no longer included unless you have one set in AD, where it was set as the same as the upn previously when you didn't)

Reply:
See the important note here: https://docs.microsoft.com/en-us/azure/active-directory/azuread-dev/azure-ad-endpoint-comparison#token-claims

Additionally regarding `email` claim, I was talking about managed users without an email attribute set. I believe the implementation changed between v1 and v2.